app/api/ada/lib/storage/database/utils.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+
+import type {
+  lf$Database,
+  lf$Row,
+  lf$Transaction,
+  lf$schema$Table,
+} from 'lovefield';
+import { size } from 'lodash';
+import ExtendableError from 'es6-error';
+
+export async function addToTable<Insert, Row>(
+  db: lf$Database,
+  tx: lf$Transaction,
+  request: Insert,
+  tableName: string,
+): Promise<Row> {
+  const table = db.getSchema().table(tableName);
+  const newRow = table.createRow(request);
+
+  const result: Row = (await tx.attach(
+    db
+      .insertOrReplace()
+      .into(table)
+      .values(([newRow]: Array<lf$Row>))
+  ))[0];
+
+  return result;
+}
+
+export const getRowFromKey = async <T>(
+  db: lf$Database,
+  tx: lf$Transaction,
+  key: number,
+  tableName: string,
+  keyRowName: string,
+): Promise<T | void> => {
```

Comment:
Any reason why you prefer void rather than null?

Reply:
`null` is a valid value in a table so you wouldn't be able to differentiate no result vs null result.